healthcheck.go

```diff
@@ -53,6 +55,71 @@ type ClusterHealth struct {
 	Status string `json:"status"`
 }
 
+//indexcheck calls elasticsearch to check if an index from the client exists
+//What is a context?
+func (cli *Client) indexcheck(ctx context.Context) (code int, err error) {
+
+	var clientElasticSearchIndices = cli.indices
+
+	for _, index := range clientElasticSearchIndices {
+
+		urlIndex := cli.url + "/" + index
+		logData := log.Data{"url": urlIndex}
+
+		URL, err := url.Parse(urlIndex)
+		if err != nil {
+			log.Event(ctx, "failed to create url for elasticsearch indexcheck", log.ERROR, logData, log.Error(err))
+			return 500, err
+		}
+
+		path := URL.String()
+		//What do the ["url"] brackets mean - is it resetting the value for the "url" key to the path
+		logData["url"] = path
+
+		//what is background context?
+		req, err := http.NewRequest("HEAD", path, nil)
+		if err != nil {
+			log.Event(ctx, "failed to create request for indexcheck call to elasticsearch", log.ERROR, logData, log.Error(err))
+			return 500, err
+		}
+
+		//Where do the credentials come from?
+		if cli.signRequests {
+			awsauth.Sign(req)
+		}
+
+		resp, err := cli.httpCli.Do(ctx, req)
+		if err != nil {
+			log.Event(ctx, "failed to call elasticsearch", log.ERROR, logData, log.Error(err))
+			return 500, err
+		}
+		defer resp.Body.Close()
+		logData["http_code"] = resp.StatusCode
+
+		switch resp.StatusCode {
+
+		case 200:
+			return resp.StatusCode, nil
```

Comment:
I think you shouldn't return here, just keep track that this index is valid, and continue the for loop. If the for loop completes, it means that all the indices have been successfully checked.

Reply:
Ah yes

Reply:
Really happy to have got feedback : ) thank you